### PR TITLE
Hotreload Implementation

### PR DESCRIPTION
**Pull request by Flowaria**

### CODE REVIEW

File: ExtraEnemyCustomization/EntryPoint.cs [hunk 1]
```diff
@@ -35,6 +35,8 @@ public override void Load()
 
             Logger.LogInstance = Log;
 
+            var useLiveEdit = Config.Bind(new ConfigDefinition("General", "Live Edit"), false, new ConfigDescription("Reload Config when they are edited while in-game"));
+            var linkMTFOHotReload = Config.Bind(new ConfigDefinition("General", "Reload on MTFO HotReload"), false, new ConfigDescription("Reload Configs when MTFO's HotReload button has pressed?"));
```
Comment (Kasuromi) [inline ExtraEnemyCustomization/EntryPoint.cs:39]: Make it active by default
```suggestion
            var linkMTFOHotReload = Config.Bind(new ConfigDefinition("General", "Reload on MTFO HotReload"), true, new ConfigDescription("Reload Configs when MTFO's HotReload button has pressed?"));
```

File: ExtraEnemyCustomization/Managers/ConfigManager__Instance.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        public GlobalConfig Global { get => GetConfig<GlobalConfig>(); }
+        public CategoryConfig Categories { get => GetConfig<CategoryConfig>(); }
+        public ModelCustomConfig ModelCustom { get => GetConfig<ModelCustomConfig>(); }
+        public AbilityCustomConfig AbilityCustom { get => GetConfig<AbilityCustomConfig>(); }
+        public ProjectileCustomConfig ProjectileCustom { get => GetConfig<ProjectileCustomConfig>(); }
+        public TentacleCustomConfig TentacleCustom { get => GetConfig<TentacleCustomConfig>(); }
+        public DetectionCustomConfig DetectionCustom { get => GetConfig<DetectionCustomConfig>(); }
+        public SpawnCostCustomConfig SpawnCostCustom { get => GetConfig<SpawnCostCustomConfig>(); }
+        public EnemyAbilityCustomConfig EnemyAbilityCustom { get => GetConfig<EnemyAbilityCustomConfig>(); }
```
Comment (Kasuromi) [inline ExtraEnemyCustomization/Managers/ConfigManager__Instance.cs:22]:
```suggestion
        public GlobalConfig Global => GetConfig<GlobalConfig>();
        public CategoryConfig Categories => GetConfig<CategoryConfig>();
        public ModelCustomConfig ModelCustom => GetConfig<ModelCustomConfig>();
        public AbilityCustomConfig AbilityCustom => GetConfig<AbilityCustomConfig>();
        public ProjectileCustomConfig ProjectileCustom => GetConfig<ProjectileCustomConfig>();
        public TentacleCustomConfig TentacleCustom => GetConfig<TentacleCustomConfig>();
        public DetectionCustomConfig DetectionCustom => GetConfig<DetectionCustomConfig>();
        public SpawnCostCustomConfig SpawnCostCustom => GetConfig<SpawnCostCustomConfig>();
        public EnemyAbilityCustomConfig EnemyAbilityCustom => GetConfig<EnemyAbilityCustomConfig>();
```

File: ExtraEnemyCustomization/Utils/Integrations/Inject/Inject_CM_PageRundown_New.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+            object hotreloader = MTFOUtil.HotReloaderField?.GetValue(null) ?? null;
+            if (hotreloader is null)
+                return;
+
+            var buttonField = hotreloader.GetType().GetField("button", BindingFlags.NonPublic | BindingFlags.Instance);
+            if (buttonField is null)
+                return;
+
+            var button = (CM_Item)buttonField.GetValue(hotreloader);
+            button.add_OnBtnPressCallback((Action<int>)MTFOUtil.OnHotReloaded);
```
Comment (Kasuromi) [inline ExtraEnemyCustomization/Utils/Integrations/Inject/Inject_CM_PageRundown_New.cs:24]: Is there no better way to do this? I have a feeling that we should have implemented it on the MTFO API side.

If not, keep it and we can PR it into MTFO

**Review by Kasuromi — CHANGES_REQUESTED** [review]

**Comment by Kasuromi** [comment]
As for "Regeneration Projectiles Somehow Crash the Game", any idea why that might be the case?

File: ExtraEnemyCustomization/Utils/Integrations/Inject/Inject_CM_PageRundown_New.cs (hunk 3 above)
Comment (Flowaria) [reply ExtraEnemyCustomization/Utils/Integrations/Inject/Inject_CM_PageRundown_New.cs:24]: Yeah In indirect way, they only have method that accepts that have interface of `IHotreload`

Comment (Kasuromi) [reply ExtraEnemyCustomization/Utils/Integrations/Inject/Inject_CM_PageRundown_New.cs:24]: I see, this solution is fine but it might break in the future

**Review by Kasuromi — APPROVED** [review]
Looks good to me

**Comment by Flowaria** [comment]
> As for "Regeneration Projectiles Somehow Crash the Game", any idea why that might be the case?

Totally no idea, it happens when it Instantiate Projectile GameObject out of Existing Projectile. Only for hotreloading and it works fine on initial game loading

**Comment by Kasuromi** [comment]
> > As for "Regeneration Projectiles Somehow Crash the Game", any idea why that might be the case?
> 
> Totally no idea, it happens when it Instantiate Projectile GameObject out of Existing Projectile. Only for hotreloading and it works fine on initial game loading

Probably best to open an issue about it and I'll look into it sometime

**Pull request merged**